compiler/rustc_middle/src/ty/fast_reject.rs

```diff
@@ -27,9 +28,12 @@ where
     UintSimplifiedType(ty::UintTy),
     FloatSimplifiedType(ty::FloatTy),
     AdtSimplifiedType(D),
+    ForeignSimplifiedType(DefId),
```

Comment:
(partially just for myself to check that later)

why do we use a `DefId` here instead of `D`? Seems like it's either unnecessary to have this be generic at all or this is a incremental bug

Reply:
jup, that's an incremental bug